### PR TITLE
Migrate suggestions to sapphire

### LINKED ISSUE #160: migrate suggestions to sapphire

### PR DESCRIPTION
**Pull request by cneryx**
# Related Issues
Resolves #160 

# Summary of Changes 
Migration

# Steps to Reproduce
Make sure previous functionality stays the same

### CODE REVIEW

File: src/commands/suggestions/suggest.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        //Confirm suggestion was taken
+        return message.reply('Codey has received your suggestion.');
+    }
+
+    private async getSuggestionDisplayInfo(suggestion: Suggestion) {
+        return (
+          '**' +
+          suggestion['id'] +
+          '** | ' +
+          suggestionStatesReadable[suggestion['state']] +
```
Comment (alexz429) [inline src/commands/suggestions/suggest.ts:38]: Probably best to use a format string here

**Review by alexz429 — CHANGES_REQUESTED** [review]

File: src/commands/suggestions/suggest.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import { Message, MessageEmbed } from 'discord.js';
+import { Args, container} from '@sapphire/framework';
+import { SubCommandPluginCommand, SubCommandPluginCommandOptions } from '@sapphire/plugin-subcommands';
+import { addSuggestion, getSuggestions, suggestionStatesReadable, SuggestionState, updateSuggestionState, Suggestion, getAvailableStatesString } from '../../components/suggestions';
+import { EMBED_COLOUR } from '../../utils/embeds';
+import _ from 'lodash';
+
+const RESULTS_PER_PAGE = 6;
```
Comment (marko-polo-cheno) [inline src/commands/suggestions/suggest.ts:9]: can u change this to 15
6 is too little and i wanna see more suggestions

File: src/components/suggestions.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      UPDATE suggestions
+      SET state = ?
+      WHERE id = ?;
+      `,
+      [state, id]
+    );
+  });
+};
+
+export const getSuggestions = async (state: string | null): Promise<Suggestion[]> => {
```
Comment (marko-polo-cheno) [inline src/components/suggestions.ts:71]: if bored, there's an issue for making getSuggById

**Review by marko-polo-cheno — APPROVED** [review]

**Review by Picowchew — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -22,17 +22,18 @@
...
-    "@sapphire/plugin-subcommands": "^2.1.3",
-    "@sapphire/utilities": "^3.3.0",
+    "@sapphire/plugin-subcommands": "^2.2.0",
+    "@sapphire/utilities": "^3.6.2",
     "concurrently": "^6.1.0",
     "cron": "^1.8.2",
     "discord.js": "^13.6.0",
     "dotenv": "^8.2.0",
     "engine-blackjack-ts": "^0.9.11",
+    "install": "^0.13.0",
```
[the remark this answers is not in the document]
Comment (marko-polo-cheno) [reply package.json:32]: tru

File: src/commands/suggestions/suggest.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+@ApplyOptions<SubCommandPluginCommandOptions>({
+    aliases: ['suggestions'],
+    description: 'Handles suggestion functions',
+    detailedDescription: `This command will forward a suggestion to the CSC executives. \
```
[the remark this answers is not in the document]
Comment (marko-polo-cheno) [reply src/commands/suggestions/suggest.ts:14]: agree

File: src/commands/suggestions/suggest.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import _ from 'lodash';
+
+const RESULTS_PER_PAGE = 15;
+
+@ApplyOptions<SubCommandPluginCommandOptions>({
+    aliases: ['suggestions'],
+    description: 'Handles suggestion functions',
+    detailedDescription: `This command will forward a suggestion to the CSC Discord Mods. \n
+    Please note that your suggestion is not anonymous, your Discord username and ID will be recorded. \n
+    If you don't want to make a suggestion in public, you could use this command via a DM to Codey instead. \n
```
Comment (Picowchew) [inline src/commands/suggestions/suggest.ts:16]: Probably remove the space before `\n`?

File: src/commands/suggestions/suggest.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import { EMBED_COLOUR } from '../../utils/embeds';
+import _ from 'lodash';
+
+const RESULTS_PER_PAGE = 15;
+
+@ApplyOptions<SubCommandPluginCommandOptions>({
+    aliases: ['suggestions'],
+    description: 'Handles suggestion functions',
+    detailedDescription: `This command will forward a suggestion to the CSC Discord Mods. \n
+    Please note that your suggestion is not anonymous, your Discord username and ID will be recorded. \n
```
Comment (Picowchew) [inline src/commands/suggestions/suggest.ts:15]: For these two lines, I think revert `\n` back to `\` so they will be on the same line?

File: src/commands/suggestions/suggest.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+const RESULTS_PER_PAGE = 6;
+
+@ApplyOptions<SubCommandPluginCommandOptions>({
+    aliases: ['suggestions'],
+    description: 'Handles suggestion functions',
+    detailedDescription: `This command will forward a suggestion to the CSC executives. \
+    Please note that your suggestion is not anonymous, your Discord username and ID will be recorded. \
+    If you don't want to make a suggestion in public, you could use this command via a DM to Codey instead. \
+    **Examples:**\n
+    \`${container.botPrefix}suggestion I want a new discord channel named #hobbies\`\n`,
```
[the remark this answers is not in the document]
Comment (Picowchew) [reply src/commands/suggestions/suggest.ts:18]: I don't think this has been addressed yet?

**Review by Picowchew — APPROVED** [review]

File: src/commands/suggestions/suggest.ts (hunk 7 above)
Comment (cneryx) [reply src/commands/suggestions/suggest.ts:15]: oops for some reason my changes disappeared

**Review by alexz429 — APPROVED** [review]

**Pull request merged**